src/crypto-api/CryptoEvent.ts

```diff
@@ -90,4 +90,36 @@ export enum CryptoEvent {
      * `progress === total === -1`.
      */
     LegacyCryptoStoreMigrationProgress = "crypto.legacyCryptoStoreMigrationProgress",
+
+    /** Fires when a new dehydrated device is created locally. */
+    DehydratedDeviceCreated = "dehydration.DehydratedDeviceCreated",
+
+    /** Fires when a new dehydrated device is successfully uploaded to the server. */
+    DehydratedDeviceUploaded = "dehydration.DehydratedDeviceUploaded",
+
+    /** Fires when rehydration has started. */
+    RehydrationStarted = "dehydration.RehydrationStarted",
+
+    /** Fires when rehydration has finished. */
+    RehydrationEnded = "dehydration.RehydrationEnded",
```

Comment:
maybe list this after `RehydrationProgress`?

Presumably it indicates that rehydration *successfully* completed? Maybe say so in the comment?

Also `RehydrationCompleted` might be a better name?